web-app/js/portal/cart/DownloadPanelItemTemplate.js

```diff
@@ -86,10 +86,33 @@ Portal.cart.DownloadPanelItemTemplate = Ext.extend(Ext.XTemplate, {
     },
 
     _downloadButton: function(collection) {
-        this._createDownloadButton.defer(1, this, [collection]);
+
+        if (collection.downloadStatus == 'requested') {
+            this._createDownloadingLabel.defer(1, this, [collection]);
+        }
+        else {
+            this._createDownloadButton.defer(1, this, [collection]);
+        }
+
         return '';
     },
 
+    // It's *actually* a button, but we're using it as a label here...
+    _createDownloadingLabel: function(collection) {
+        var elementId = this._getButtonId(collection, 'downloadButtonId');
+
+        Ext.fly(elementId).update("");
+
+        new Ext.Button({
+            text: OpenLayers.i18n('downloadStatusRequested'),
+            icon: 'images/spinner_orange_background.gif',
```

Comment:
You could use Font Awsome. Thats the purely css way, but the FA spinner is a bit jumpy looking. The class would most likely be 'fa-spinner' from (http://fortawesome.github.io/Font-Awesome/cheatsheet/)

Examples of this is in Portal.search.FacetedSearchResultsDataView
```
    _buildLabel: function(fontAwesomeClass, text) {
        return "<span class=\"fa fa-fw " + fontAwesomeClass + "\"></span> " + text;
    },
```